gazebo_ros/test/disable_perf_metrics.test.py

```diff
@@ -0,0 +1,126 @@
+# limitations under the License.
+
+import unittest
+
+import launch
+import launch.actions
+
+import launch_testing
+import launch_testing.asserts
+import pytest
+import os
+
+import rclpy
+from gazebo_msgs.msg import PerformanceMetrics
+from rclpy.node import Node
+from rcl_interfaces.msg import Parameter, ParameterValue
+from ros2param.api import call_set_parameters
+import time
+from threading import Thread
+
+
+@pytest.mark.launch_test
+def generate_test_description():
+    os.chdir(os.path.dirname(__file__))
+    return launch.LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so'],
+            output='screen',
+            sigterm_timeout='20',
+            sigkill_timeout='20'
+        ),
+        launch_testing.actions.ReadyToTest()
+    ])
+
+
+class MyTestingClass(unittest.TestCase):
+    def test_routine(self):
+        rclpy.init()
+        node = MakeTestNode()
+        node_found = node.wait_for_gazebo_node()
+
+        if node_found:
+            response = node.set_parameter(False)
+            if response.successful:
+                print("enable_performance_metrics successfully set to false !")
+                node.start_subscriber()
+                time.sleep(2.5)
+
+                if node.msg_count > 0:
+                    print("Received {} messages, test failed".format(node.msg_count))
+                    rclpy.shutdown()
+                    assert False
+                else:
+                    print("No messages were published")
+                    rclpy.shutdown()
+                    assert True
+            else:
+                print("Parameter could not be set, exiting")
+                rclpy.shutdown()
+                assert False
+        else:
+            print("Could not find Gazebo node, exiting")
+            rclpy.shutdown()
+            assert False
+
+
+class MakeTestNode(Node):
+    def __init__(self):
+        super().__init__('testing_node')
+        self.flag_gazebo_node = False
+        self.msg_count = 0
+
+    def wait_for_gazebo_node(self):
+        start = time.time()
+        print("Waiting to find Gazebo Node...")
+        while time.time() - start < 5 and not self.flag_gazebo_node:
+            active_nodes = self.get_node_names()
+            if 'gazebo' in active_nodes:
+                self.flag_gazebo_node = True
+                print("Found the gazebo node !")
+                time.sleep(0.1)
+
+        if not self.flag_gazebo_node:
+            print("Gazebo node was not found !")
+        return self.flag_gazebo_node
+
+    # Set enable_performance_metrics parameter to false
+    def set_parameter(self, state=False):
+        parameter = Parameter()
+        parameter.name = 'enable_performance_metrics'
+
+        temp_value = ParameterValue()
+        temp_value.bool_value = state
+        temp_value.type = 1
+        parameter.value = temp_value
```

Comment:
I think we can avoid the temporary. Also, prefer using the named constants defined in the message instead of literal numbers:
```suggestion
        parameter.value.bool_value = state
        parameter.value.type = ParameterType.PARAMETER_BOOL
```